subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -103,6 +106,11 @@ static char delete_rejected_topic[DELETE_REJECTED_TOPIC_LEN + 1];
 /* Empty string used to request the AWS IoT shadow document. */
 #define AWS_IOT_SHADOW_REQUEST_STRING ""
 
+#define AWS_BROKER_HOST_NAME_PREFIX "%s"
```

Comment:
You could remove this macro and just use `%s` in place.

Same goes for `CONFIG_AWS_IOT_BROKER_HOST_NAME_MAX_LEN`. Just declare:
`static char aws_host_name_buf[CONFIG_AWS_IOT_BROKER_HOST_NAME_MAX_LEN + 1];` directly.